### PR TITLE
Publish @autorest/csharp-v3 on npm

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/autorest.csharp/issues/851

### CODE REVIEW

File: eng/PublishRelease.ps1 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    npm version --no-git-tag-version $devVersion | Out-Null;
+   
+    $file = npm pack -q;
+    $name = "Autorest for c# v$devVersion"
+
+    Write-Host "Publishing $file on GitHub!"
+    
+    cmd /c ""npx -q publish-release --token $GitHubToken --repo autorest.csharp --owner azure --name $name --tag v$devVersion --notes=prerelease-build --prerelease --editRelease false --assets $file --target_commitish $Sha 2>&1""
+
+    $filePath = Join-Path $WorkingDirectory '.npmrc'
```
Comment (pakrym) [inline eng/PublishRelease.ps1:24]: Should we use the recommedned pattern of checking in the npmrc and setting the token as an env var: https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#set-the-token-as-an-environment-variable-on-the-cicd-server

File: eng/PublishRelease.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+try {
+    $currentVersion = node -p -e "require('./package.json').version";
+    $devVersion = "$currentVersion-dev.$BuildNumber"
+
+    Write-Host "Setting version to $devVersion"
+
+    npm version --no-git-tag-version $devVersion | Out-Null;
+   
+    $file = npm pack -q;
+    $name = "Autorest for c# v$devVersion"
```
Comment (pakrym) [inline eng/PublishRelease.ps1:18]:
```suggestion
    $name = "AutoRest C# v$devVersion"
```

**Review by pakrym — APPROVED** [review]

**Comment by pakrym** [comment]
I've looked at the npm package and it seems to contain too much stuff in it. There are two copies of everything.

File: eng/PublishRelease.ps1 (hunk 1 above)
Comment (pakrym) [reply eng/PublishRelease.ps1:24]: I don't feel strongly.

**Comment by ShivangiReja** [comment]
>I've looked at the npm package and it seems to contain too much stuff in it. There are two copies of everything.

Removed `published` folder with all the duplicate files in it.

Here is the latest NPM release: https://www.npmjs.com/package/@autorest/csharp-v3/v/3.0.0-beta.20201216.11
Here is the latest GitHub release: https://github.com/Azure/autorest.csharp/releases/tag/v3.0.0-beta.20201216.11

**Review by pakrym — APPROVED** [review]
👍 👍 👍

**Pull request merged**